config/routes.rb

```diff
@@ -1,52 +1,63 @@
 Rails.application.routes.draw do
   root "welcome#index"
-  get "events/new"
-  post "/events/:event_id/manage_staffs" => "manager_profiles#save_staffs",
-       as: :save_staffs
-  get "/events/:event_id/manage_staffs" => "manager_profiles#manage_staffs",
-      as: :manage_staffs
-  get "/events/:event_id/remove_staff/:event_staff_id" =>
-    "manager_profiles#remove_staff", as: :remove_staff
-  get "/events/:id/enable" =>
-          "events#enable", as: :enable_event
-  get "/events/:id/disable" =>
-          "events#disable", as: :disable_event
-  get "/events/:id/generate" => "events#generate", as: :generate_event
-  get "/featured_events" => "welcome#featured"
-  get "/popular_events" => "welcome#popular"
-  match "/events/popular" => "events#popular", via: [:post, :get]
-  get "/upcoming_events" => "welcome#index"
-  get "/my_events" => "users#show"
-  get "events/loading"
-  get "/lookup_staffs" => "users#lookup_staff_emails"
-  get "/user_info/:user_id" => "users#fetch_user_info"
-  post "/refund/:uniq_id" => "bookings#request_refund", as: :refund
-  post "/paypal_hook" => "bookings#paypal_hook", as: :paypal_hook
-  get "/scan_ticket" => "bookings#scan_ticket", as: :scan_ticket
-  get "/events/:id/scan" => "events#scan", as: :gatekeeper
-  get "/scan_ticket/:ticket_no" => "bookings#use_ticket", as: :scan
-  get "unattend", to: "attendees#destroy", as: "unattend"
-  get "auth/:provider/callback", to: "sessions#create"
-  get "auth/failure", to: redirect("/")
-  get "/tickets" => "bookings#index"
-  get "/events/:id/tickets" => "events#tickets", as: :event_tickets
-  get "/events/:id/tickets-report" =>
-          "events#tickets_report", as: :tickets_report
-  get "/print/:booking_id(/:ticket_type_id)" => "printer#print", as: :print
-  post "/print/:id" => "printer#redirect_to_print"
-  get "/print/:id" => "printer#redirect_to_print"
-  get "/download/:booking_id" => "printer#download", as: :download
-  get "signout", to: "sessions#destroy", as: "signout"
-  get "/session" => "sessions#create"
-  post "/api_login" => "sessions#api_login"
+
+  scope controller: :events do
+    get "/events/popular"            => :popular
+    get "/events/loading"
+    get "/events/:id/enable"         => :enable, as: :enable_event
+    get "/events/:id/disable"        => :disable, as: :disable_event
+    get "/events/:id/generate"       => :generate, as: :generate_event
+    get "/events/:id/scan"           => :scan, as: :gatekeeper
+    get "/events/:id/tickets"        => :tickets, as: :event_tickets
+    get "/events/:id/tickets-report" => :tickets_report, as: :tickets_report
```

Comment:
This can also be refactored to 

```
scope path: '/events', controller: :events do
  get 'loading' 
  get ':id/enable' => :enable, as: :enable_event
 get ':id/disable' => :disable, as: :disable_event 
end
```